### DISCUSSION TITLE
500 Internal Server Error

### DISCUSSION THREAD

**Discussion by harxish** [fastapi/fastapi#6838] [Questions] [answered]
Hey, this is my first time using FastAPI. I've started with the docs but I am not able to execute the first code snippet. The applications starts but when I go to my browser it shows me Internal Server Error. 
```

INFO:     Started server process [18501]
INFO:     Waiting for application startup.
INFO:     ASGI 'lifespan' protocol appears unsupported.
INFO:     Application startup complete.
INFO:     Uvicorn running on http://127.0.0.1:8000 (Press CTRL+C to quit)
ERROR:    Exception in ASGI application
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/uvicorn/protocols/http/httptools_impl.py", line 385, in run_asgi
    result = await app(self.scope, self.receive, self.send)
  File "/usr/lib/python3/dist-packages/uvicorn/middleware/proxy_headers.py", line 45, in __call__
    return await self.app(scope, receive, send)
TypeError: root() takes 0 positional arguments but 3 were given
INFO:     127.0.0.1:34956 - "GET / HTTP/1.1" 500 Internal Server Error
```

FastAPI version : 0.63.0
Python version : 3.8.6

**Comment by insomnes** [comment]
@harxish hello! Can you please provide the snippet or link to it?

**Comment by harxish** [comment]
Code : 

```
from fastapi import FastAPI

app = FastAPI()


@app.get("/")
async def root():
    return {"message": "Hello World"}

```

**Comment by insomnes** [comment, marked as answer]
@harxish are you sure, that you are starting uvicorn with proper command in this case?
The command in [docs](https://fastapi.tiangolo.com/#run-it) is:
`uvicorn main:app --reload`
This log is unusual for proper uvicorn start:
`INFO:     ASGI 'lifespan' protocol appears unsupported.`

**Comment by harxish** [comment]
Oops.... My bad, I gave in root instead of app :(

Thanks.

**Comment by MicoAye13** [comment]
Hi! Im having the same problem, i can run message:hello world after doing a post request I'm getting internal server error

**Comment by harxish** [comment]
@Micoaye0610 Can you share your code

**Comment by suelopoder** [comment]
For a next visitor that's confused as I was:

That last line: `TypeError: root() takes 0 positional arguments but 3 were given`
Refers to your code, not fastapi.
Probably your `main.py` where you have a `root()` function that needs some extra arguments.

For some reason this was working on my machine but not on a docker python image.
